### PR TITLE
fix: liquidjs login routes

### PR DESCRIPTION
**Pull request by danielgent**
Will allow me to do sign-ups much more easily to fix the web component tests :partying_face:

### CODE REVIEW

File: src/routes/tsoa/login.ts [hunk 1]
```diff
@@ -99,7 +99,7 @@ export class LoginController extends Controller {
       throw new HTTPException(400, { message: "Session not found" });
     }
 
-    return renderLogin(env, this, session);
+    return renderLogin(env, this, session, state);
```
[the remark this answers is not in the document]
Comment (danielgent) [reply src/routes/tsoa/login.ts:102]: We could even parse the output into JS dom and then assert we can navigate around

File: src/routes/tsoa/login.ts [hunk 2]
```diff
@@ -422,7 +422,7 @@ export class LoginController extends Controller {
   ): Promise<string> {
     const { env } = request.ctx;
     const session = await env.data.universalLoginSessions.get(state);
-    if (!session?.username) {
+    if (!session) {
```
Comment (danielgent) [inline src/routes/tsoa/login.ts:425]: not sure what the previous reference was for!

File: src/templates/render.ts [hunk 3]
```diff
@@ -27,10 +27,10 @@ export async function renderForgotPassword(
   controller.setHeader("content-type", "text/html");
   controller.setStatus(200);
 
-  const content = await engine.render(template, context);
+  const content = await engine.render(template, { ...context, state });
   return engine.render(layoutTemplate, {
     ...context,
-    state,
+    state, // seems inconsistent doing this here... does it need doing in both places?
```
Comment (danielgent) [inline src/templates/render.ts:33]: does `layoutTemplate` need state? doesn't look like it...

If not - I'll remove it everywhere

if so - I can pass it in everywhere

File: src/templates/render.ts [hunk 4]
```diff
@@ -78,7 +79,9 @@ export async function renderLogin(
   const content = await engine.render(template, {
     ...context,
     connections,
+    // in some places this is in the template render, in others it's in the layout renderer... which is it?
     errorMessage,
```
Comment (danielgent) [inline src/templates/render.ts:83]: same with `errorMessage` - is this needed in both templates?

File: src/templates/render.ts (hunk 3 above)
Comment (markusahlstrand) [reply src/templates/render.ts:33]: Yes.. I'm not sure if state is part of that object.. Started to pass it separately but it's probably not really consistent.

File: src/templates/render.ts (hunk 4 above)
Comment (markusahlstrand) [reply src/templates/render.ts:83]: I tried moving error message out to a separate param.

**Review by markusahlstrand — APPROVED** [review]
This this is a step in the right direction. Probably a few more things to clean up.

**Pull request merged**